dev-debug/bpftrace/bpftrace-0.24.1.ebuild

```diff
@@ -115,18 +115,24 @@ src_configure() {
 	local mycmakeargs=(
 		# DO NOT build the internal libs as shared
 		-DBUILD_SHARED_LIBS=OFF
-		# DO dynamically link the bpftrace executable
-		-DSTATIC_LINKING:BOOL=OFF
-		-DBUILD_TESTING:BOOL=$(usex test)
+		-DBUILD_TESTING=$(usex test)
+		-DBUILD_UBSAN=$(usex sanitize)
 		# we use the pregenerated man page
-		-DENABLE_MAN:BOOL=OFF
-		-DENABLE_SYSTEMD:BOOL=$(usex systemd)
-		-DENABLE_SKB_OUTPUT:BOOL=$(usex pcap)
+		-DENABLE_MAN=OFF
+		-DENABLE_SKB_OUTPUT=$(usex pcap)
+		-DENABLE_SYSTEMD=$(usex systemd)
+		# DO dynamically link the bpftrace executable
+		-DSTATIC_LINKING=OFF
 	)
 
 	cmake_src_configure
 }
 
+src_test() {
+	use sanitize && export UBSAN_OPTIONS="print_stacktrace=1:halt_on_error=1"
```

Comment:
I'd rather we encourage people to use https://wiki.gentoo.org/wiki/UndefinedBehaviorSanitizer appropriately as an env.

But we can do _something_ here. If naively following that for *FLAGS etc doesn't work, I suggest copying what I've done in e.g. the botan, or elfutils ebuilds, rather than adding a USE flag.

Reply:
Yeah I wasn't really sure if that's appropriate, or whether to couple test+sanitize or keep them separate. I also didn't find a good way to check if the toolchain has sanitizers enabled so I just ignored that for now.
The USE flag is more or less required because it must be passed to the cmake build, because of upstream reasons (read the convoluted thread in the ubsan patch PR).

Reply:
Sounds good. Then just filter it from flags if needed too.

Reply:
Filtering not needed, it's just that they insisted on making things complicated by selectively not building some things (tests) with -fsanitize. If nd when they get their ASAN sorted we can revisit this if necessary.